Sofa/framework/Type/src/sofa/type/Mat.h

```diff
@@ -1137,13 +1137,19 @@ void printMaple(std::ostream& o, const Mat<L,C,real>& m)
 
 
 /// Create a matrix as \f$ u v^T \f$
-template <sofa::Size L, sofa::Size C, typename T>
-constexpr Mat<L,C,T> dyad( const Vec<L,T>& u, const Vec<C,T>& v ) noexcept
+template <class Tu, class Tv>
+constexpr Mat<Tu::size(), Tv::size(), typename Tu::value_type>
+dyad(const Tu& u, const Tv& v) noexcept
 {
-    Mat<L,C,T> res(NOINIT);
-    for(sofa::Size i=0; i<L; i++ )
-        for(sofa::Size j=0; j<C; j++ )
-            res[i][j] = u[i]*v[j];
+    static_assert(std::is_same_v<typename Tu::value_type, typename Tv::value_type>);
```

Comment:
Templates have changed so this PR should be breaking unfortunately.
It could need deprecation (etc.) 😬 but personally I dont think it is mandatory to do... 🫡
The two problematic cases seem quite unlikely:
 - Mat::dyad is overridden 
 - one did use dyad with explicit templates